docs/specs/user-profile/design.md

```diff
@@ -147,49 +147,98 @@ In addition to standard attributes, the developer can define their own custom at
 
 ### Defining custom attributes
 
-The custom attributes are defined as a single JSON schema written against a subset of JSON schema 2019-09.
+The custom attributes are defined with a configuration.
 
-Here is an example of the schema of the custom attributes.
+Here is an example of the configuration.
 
-```yaml
+```
 user_profile:
   custom_attributes:
-    schema:
-      properties:
-        stripe_customer_id:
-          type: string
-        hobby:
-          type: string
+    attributes:
+    - pointer: /x_phone_number
+      type: phone_number
+    - pointer: /x_email
+      type: email
 ```
 
-All changes made to custom attributes must be valid against the schema.
+Each custom attribute MUST have an unique pointer and of one of the defined types.
+
+The pointer of custom attribute MUST have exactly ONE level, as in `/this_is_one_level`.
+The pointer MUST also be non-empty, so `/` is not a valid pointer.
+
+The pointer MUST NOT conflict with the pointer of any standard attributes,
+so the developer CANNOT define a custom attribute with pointer `/email`.
+
+NO data migration is ever done when the configuration is changed.
+This means if the developer change the pointer, existing data WILL BE lost.
 
-#### Supported subset of JSON schema 2019-09
+The label of the custom attribute can be localized with the translation key `custom-attribute-label-{pointer}`.
+For example, if the custom attribute is `/x_email`, then the translation key is `custom-attribute-label-/x_email`.
 
-- `type`: `boolean`, `string`, `number`, `integer`
-- `format`: `email`, `phone`, `uri`, `date-time`
-- `enum`
-- `multipleOf`
-- `maximum`
-- `exclusiveMaximum`
-- `minimum`
-- `exclusiveMinimum`
-- `maxLength`
-- `minLength`
-- `properties`
+#### Custom Attribute type `string`
 
-### Editing custom attributes in the settings page and in the portal
+The custom attribute is of type string.
+The UI control of it is a text field.
 
-The UI control of the custom attribute is determined by the `type` and the `format`.
+#### Custom Attribute type `integer`
+
+The custom attribute is of type integer.
+Optionally, the developer can define the minimum and the maximum allowed value.
+
+For example, if the developer wants to define a custom attribute of non-negative integer, they write
+
+```
+- pointer: /x_age
+  type: integer
```

Comment:
Will we support number in this stage? Or later?

Reply:
Let me add `number` too.